crates/core/component/fee/src/component.rs

```diff
@@ -35,12 +40,27 @@ impl Component for Fee {
     ) {
     }
 
-    #[instrument(name = "fee", skip(_state, _end_block))]
+    #[instrument(name = "fee", skip(state, _end_block))]
     async fn end_block<S: StateWrite + 'static>(
-        _state: &mut Arc<S>,
+        state: &mut Arc<S>,
         _end_block: &abci::request::EndBlock,
     ) {
-        // TODO: update gas prices here eventually
+        let state_ref = Arc::get_mut(state).expect("unique ref in end_block");
+        // Grab the total fees and use them to emit an event.
+        let fees = state_ref.accumulated_base_fees_and_tips();
```

Comment:
If we don't take the fees out of the state here, won't they be present in the next block, causing fees to just accumulate over time?

Reply:
This is using ephemeral storage, so the tally is discarded after the block finish executing.

Reply:
Also the fees have to be taken out to prevent duplication of value